js/ui/windowManager.js

```diff
@@ -2268,17 +2268,26 @@ const CodingManager = new Lang.Class({
         dst.opacity = 0;
 
         let srcGeometry = src.meta_window.get_frame_rect();
+        let dstGeometry = dst.meta_window.get_frame_rect();
+
+        if (srcGeometry.equal(dstGeometry))
+            return;
```

Comment:
I'm not sure if the geometries being equal is enough to check here. The windows could still be of the same geometry, but one could be maximized and the other unmaximized. This code will need to check for that too before returning early.